src/ForwardDiffNumber.jl

```diff
@@ -128,6 +131,9 @@ isfinite(n::ForwardDiffNumber) = isfinite(value(n))
 isinf(n::ForwardDiffNumber) = isinf(value(n))
 isreal(n::ForwardDiffNumber) = isconstant(n)
 
+floor{T}(::Type{T}, n::ForwardDiffNumber) = floor(T, value(n))
```

Comment:
Maybe in this case it's okay because the derivative of floor is zero almost everywhere anyway.